windows/windows_update_install/install_online_updates.yml

```diff
@@ -1,23 +1,34 @@
 # Copyright 2024 VMware, Inc.
 # SPDX-License-Identifier: BSD-2-Clause
-#
+---
+- name: "Initialize Windows online updates install status"
+  ansible.builtin.set_fact:
+    win_online_updates_installed: false
+
 - name: "Get guest OS version build"
   include_tasks: ../utils/win_get_os_version.yml
-- name: "Display the guest OS version build before online Windows Updates"
-  ansible.builtin.debug:
-    msg: "Before running online Windows Updates, guest OS version build is {{ win_os_version_build }}"
+- name: "Display the guest OS version build before installing Windows online updates"
+  ansible.builtin.debug: var=win_os_version_build
 
-- name: "Set the online Windows Updates reject list"
+- name: "Set the Windows online updates reject list"
   ansible.builtin.set_fact:
     win_updates_reject_list: "{{ windows_updates_reject_list.split(',') | map('trim') }}"
   when:
     - windows_updates_reject_list is defined
     - windows_updates_reject_list
 
-- name: "Install online Windows Updates"
+- name: "Get Windows online updates"
+  include_tasks: ../utils/win_get_online_updates.yml
+  vars:
+    win_updates_ignore_errors: true
+
+- name: "Install Windows online updates"
   include_tasks: ../utils/win_install_online_updates.yml
   vars:
     win_updates_ignore_errors: true
+  when:
+    - win_get_updates_result.found_update_count is defined
+    - win_get_updates_result.found_update_count | int != 0
 
 - name: "Check if online Windows Updates log file exists in guest OS"
```

Comment:
"online Windows Updates" --> "Windows online updates"

And can also update the remaining ones in other places of the file. Thanks.

Reply:
Updated, thanks.